.github/workflows/spack_build.yml

```diff
@@ -141,10 +144,13 @@ jobs:
                 padded_length: False
             mirrors:
               local-buildcache: oci://${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
-            # spack: https://binaries.spack.io/develop
+              spack: https://binaries.spack.io/develop
             packages:
               all:
                 require: "%gcc"
+              # https://spack.readthedocs.io/en/latest/packages_yaml.html#setting-requirements-on-virtual-specs
+              cmake:
+                require: "~qtgui~ncurses"
```

Comment:
@cameronrutherford how did you figure out these parameters?

Reply:
So this was part of my initial debugging.

As we are using `spack@develop`, we are always getting the latest and "greatest" changes. In this case we inherited some changes to CMake that seemed to make it build these options when it wasn't previously.

This forces any build of CMake to have these options disabled.

See [this job's](https://github.com/LLNL/hiop/actions/runs/11710960641/job/32618445922) concretization step for an example of what I mean. Compare that to a later passing pipeline [here](https://github.com/LLNL/hiop/actions/runs/11730809533/job/32679500556). Notice how the updated config results in a much smaller build.

Sure we could probably get the more complicated builds to work too, but a minimal build is faster, and less error prone - we shouldn't build things we don't need.

Reply:
What pointed me to this was noticing the delta in the concretizations. I looked through the logs, noticed that we were having builds fail on a dependency we don't normally build, and then figured out what package was to blame. Then I worked with that package's spack build config to figure out what to enable / disable.

https://packages.spack.io is a great way to get this information quickly.